lib/smart_answer_flows/pay-leave-for-parents.rb

```diff
@@ -138,7 +138,42 @@ def define
         end
 
         next_node do
-          outcome :mother_earned_at_least_390
+          if %w[employee self-employed].include?(calculator.employment_status_of_mother) && calculator.mother_worked_at_least_26_weeks == "no"
```

Comment:
There are a lot of changes in this commit (0c41710492bdd7546d43614052ec37306456b534) - where did all this code come from? How can we be sure all these paths are tested?

Also, nitpick: the 1-line commit message is too long for GitHub's UI, could it be rephrased?

Reply:
This question used to always direct to `mother_earned_at_least_390` question - however for users under certain conditions who had answered "no" the input from `mother_earned_at_least_390` was irrelevant as their next step was already determined. So we are now able to direct users to their outcome as soon as possible.

This means this new code block and the existing `mother_earned_at_least_390`code is fairly similar.

https://github.com/alphagov/smart-answers/blob/0c41710492bdd7546d43614052ec37306456b534/lib/smart_answer_flows/pay-leave-for-parents.rb#L190

As the only change to the flow is missing out a question under certain conditions, the existing tests with those conditions failed. Removing the now missing question from the tests resolved the issue.

Reply:
I've also update the commit message with some more detail on this

Reply:
I've been having a think about this and have worked up a spike here: https://github.com/alphagov/smart-answers/pull/5254

Whilst the spike makes the code a bit more verbose (and does repeat some of the logic, as you suggested it might), I think it's a lot easier to follow and is probably a safer / more maintainable approach to working with Smart Answers from now on.

What do you think? If you like it, I'm tempted to write it up as an ADR for the repo. Either way, I'm happy to approve this PR, it looks good 👍

Reply:
I think this looks like a sensible approach - I'd suggest it's outside the scope of this PR but it's certainly worth refactoring the whole flow once this PR has been merged, and I'd be happy to take it on